### PR TITLE
Apply global media settings except on product page

### LINKED ISSUE #1083: [Global setting] Media corner radius bug
### Expectation

- [x] Have the border follow [the media corner radius setting](https://screenshot.click/22-01-nad48-c572p.png)

### PR DESCRIPTION
**Pull request by martinamarien**
**Why are these changes introduced?**
This PR applies the media global settings to the sections/features of the theme (excluding product page and featured product).

I will address product page media in a separate PR. 

Fixes #1083 
Fixes some items in #928

**What approach did you take?**
The styling was being re-applied to many different features/classes, so I created a utility class that can be applied to all media items. 

This is similar to the approach taken for content-containers

Applies to: 
- [x] Video section
- [x] Collage video
- [x] Collage image
- [x] Cart notification media
- [x] Cart item media (on cart page) 
- [x] Image with text media 

**Other Considerations**
I updated the slideshow class section to use "Section" so that it can inherit the section spacing global settings. 

**Feedback requested**
Questions for Meli/Reviewers: 

Should these styles also apply to: 
- [x] Media on page templates
- [ ] ~Image media in Footer~
- [ ] Slideshow media (separate issue)

**Demo links**
- [Editor](https://os2-demo.myshopify.com/admin/themes/127063719958/editor)

**Checklist**
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] N/A Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by melissaperreault** [comment]
> Questions for Meli/Reviewers:
> Should these styles also apply to:

- Media on page templates: _**YES**_
- Image media in Footer: _**NO**_, I would keep it low for now on footer as there could be various layout and needs.
- Slideshow media: _**MAYBE**_, To discuss the details with @nicklepine and @danielvan 🙏 I will report back!

File: assets/base.css [hunk 1]
```diff
@@ -2519,7 +2519,33 @@ details-disclosure > details {
   }
 }
 
-.content-container--full-width {
+.global-media-settings {
```
[the remark this answers is not in the document]
Comment (martinamarien) [reply assets/base.css:2522]: Great point Ken (and Tyler). 
Changing this here will conflict with my other media branch. I think I'll opt to make a fast follow once they've both been merged and update the naming to `settings-media` and `settings-content-containers`.

**Review by melissaperreault — COMMENTED** [review]
Thank you Martina! 🎉 Looking good!

Some observations:

- [ ] Not sure if this is relevant but since we are making the Collage blocks inherit the `Cards` settings from https://github.com/Shopify/dawn/pull/1090, this won't be necessary anymore for Collage `Video` ad `Image` block.
- [ ] Omit shadow settings, for `Cart template` and `Cart notification` thumbnails, like we'll do for `Product template` thumbnails.
- [ ] I wonder if we want to inherit `Media` settings on the [Collapsible content section](https://screenshot.click/22-01-bet2j-7f200.png) image until we apply `Content containers` to it from https://github.com/Shopify/dawn/issues/1043
- [ ] Not sure what we can do, when you link a page under the `Collapsible row block`, [the shadow gets cut](https://screenshot.click/22-01-494w3-4aj5q.png).

File: assets/base.css [hunk 2]
```diff
@@ -2519,7 +2516,30 @@ details-disclosure > details {
...
+  content: '';
+  position: absolute;
+  inset: calc(var(--media-border-width) * -1);
+  border-radius: var(--media-radius);
+  box-shadow: var(--media-shadow-horizontal-offset) var(--media-shadow-vertical-offset) var(--media-shadow-blur-radius) rgba(var(--color-shadow), var(--media-shadow-opacity));
+  z-index: -1;
+}
+
+.global-media-settings img,
+.global-media-settings iframe {
```
[the remark this answers is not in the document]
Comment (melissaperreault) [reply assets/base.css:2536]: No, we don't want to apply the settings for inside the modal.

File: sections/slideshow.liquid [hunk 3]
```diff
@@ -145,7 +145,7 @@
 {
   "name": "t:sections.slideshow.name",
   "tag": "section",
-  "class": "spaced-section spaced-section--full-width",
+  "class": "section",
```
[the remark this answers is not in the document]
Comment (melissaperreault) [reply sections/slideshow.liquid:148]: This PR https://github.com/Shopify/dawn/pull/1110 is taking care of adding padding setting to header and footer to space out the next section below it in relationship with other settings combo.

**Comment by martinamarien** [comment]
@kmeleta @LucasLacerdaUX @melissaperreault 
I have addressed your feedback and rebased the branch. This means that the collage section should now be using the content containers global settings. 

Meli - these two points are now fixed.  
- [x] Not sure if this is relevant but since we are making the Collage blocks inherit the Cards settings from #1090, this won't be necessary anymore for Collage Video ad Image block.
- [x] Omit shadow settings, for Cart template and Cart notification thumbnails, like we'll do for Product template thumbnails.

I will address slideshow in a separate PR.

As for the collapsible content points. I will create an issue to tackle these seperately. 
- [ ] I wonder if we want to inherit Media settings on the Collapsible content section image until we apply Content containers to it from #1043
 - [ ] Not sure what we can do, when you link a page under the Collapsible row block, the shadow gets cut.

**Review by kmeleta — COMMENTED** [review]
Just noticed one little code thing we can cleanup. Otherwise I think this is looking good to me

**Review by melissaperreault — APPROVED** [review]
Thank you Martina! 🎉 LGTM!
I made some comment below, we can ship and follow up on another PR once we have alignment on the path to move forward. 🙏 Thanks!

>  This means that the collage section should now be using the content containers global settings

- I think you meant `Cards` instead of `Content containers` global settings. 😄  Anyway, looks good!

- I also reviewed the issue from https://github.com/Shopify/dawn/issues/1099 about Image with text, looks good now 👍 

Other observations:

- [ ] I believe we might have no choice to do a systematic scale for the border for thumbnails in Cart notification and Cart template. cc. @danielvan @nicklepine 
   - Cart template
      - Mobile: the delete icon gets pushed when we increase the media border thickness. We would want to maintain the same positioning regardless of the range value. ([Video recording](https://screenshot.click/22-01-co6u0-531km.mp4))
      - Desktop: the thumbnails [are getting way too small](https://screenshot.click/22-01-dcxum-32ung.png)
   - Cart notification: [thumbnail gets small](https://screenshot.click/22-01-xnbwg-kyhvu.png) for an intent of large border on larger components.

**Comment by nicklepine** [comment]
Thanks @melissaperreault . I created https://github.com/Shopify/dawn/issues/1164 to explore the systematic border thickness adjustment for these areas. This can be a Sev 2

File: assets/component-deferred-media.css [hunk 4]
```diff
@@ -6,6 +6,12 @@
   padding: 0;
   height: 100%;
   width: 100%;
+  border-radius: calc(var(--media-radius) - var(--media-border-width));
+  overflow: hidden;
+}
+
+.global-media-settings--full-width .deferred-media__poster {
+  border-radius: 0;
```
Comment (LucasLacerdaUX) [inline assets/component-deferred-media.css:14]: Now that `Collage` cards are using settings from `Cards`, the `Media` is still causing changes to the image radius in these blocks:
image

Should this be replaced with:
```suggestion
.collage-card .deferred-media__poster {
  border-radius: 0;
```
or is this snippet accounting for something else? There's a `.collage-card .deferred-media__poster` selector in `collage.css` already if you prefer to add it there.

Comment (martinamarien) [reply assets/component-deferred-media.css:14]: ^ This was actually still an issue on 1124, so I did fix it there.

File: assets/component-image-with-text.css [hunk 5]
```diff
@@ -91,56 +91,58 @@
...
-.image-with-text:not(.image-with-text--overlap) .grid__item:last-child > *{
+.image-with-text:not(.image-with-text--overlap) .image-with-text__media-item > *,
+.image-with-text:not(.image-with-text--overlap) .image-with-text__text-item > * {
   box-shadow: none;
 }
 
 @media screen and (max-width: 749px) {
-  .image-with-text.collapse-corners:not(.image-with-text--overlap) .grid__item:first-child:after,
-  .image-with-text.collapse-corners:not(.image-with-text--overlap) .grid__item .image-with-text__media {
+  .image-with-text.collapse-corners:not(.image-with-text--overlap) .image-with-text__media-item:after,
```
Comment (LucasLacerdaUX) [inline assets/component-image-with-text.css:138]: Setting the `Border width` to `0` makes the border radius look broken on both Desktop and Mobile.
 

Here's what it looks like if I increase `Border width` to `1`:

**Review by LucasLacerdaUX — COMMENTED** [review]
The issue I mentioned above is now fixed. I think there's still a bug with the `Image with Text` section.

File: assets/component-image-with-text.css (hunk 5 above)
Comment (martinamarien) [reply assets/component-image-with-text.css:138]: We do need these styles applied, but I think we might need to revise the `collapse-corners` condition. 
That said, this is already something I can replicate on `main`, so I suggest we create an issue for this, and tackle in a separate PR

Screen Shot 2022-01-14 at 10 22 46 AM

**Review by LucasLacerdaUX — DISMISSED** [review]
🐶👍

File: assets/component-image-with-text.css (hunk 5 above)
Comment (kmeleta) [reply assets/component-image-with-text.css:138]: I think that's actually a case we wanted, though Meli and I were going to discuss it more. @melissaperreault this our weird condition right?

**Review by melissaperreault — APPROVED** [review]

**Pull request merged**